units/projectile.gd

```diff
@@ -2,5 +2,12 @@ extends Area2D
 
 const SPEED = 750
 
+
 func _physics_process(delta):
 	position += transform.x * SPEED * delta
+
+
+func _on_area_entered(area):
+	# TODO: apply effects on enemy hit. For now just despawn.
+	queue_free()
+	area.owner.queue_free()
```

Comment:
what is the `area.owner` actually, here?

Reply:
Enemy, so that we remove the whole enemy and not just the collision box.
But yes, this feels a little 🪄 doesn't it. A next step would be to add a method to the collider for handling what to do, and we could just call that as `area.handle_projectile_collision` or something.

Reply:
Doesn't `queue_free` clear the whole scene? (at least below the node?)

So instead of `owner.queue_free`, we should be emitting a "hit" event here or something, and the `Enemy` node (or whatever is the root of the scene) should handle the queue_free-ing

But that's just details. :ship: :rocket: `merge-when-approved`